dev-libs/folly/folly-0.57.0.ebuild

```diff
@@ -0,0 +1,47 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit eutils autotools
+
+DESCRIPTION="An open-source C++ library developed and used at Facebook"
+HOMEPAGE="https://github.com/facebook/folly"
+SRC_URI="https://github.com/facebook/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+static-libs"
+
+RDEPEND="
+	app-arch/lz4
+	app-arch/snappy
+	app-arch/xz-utils
+	dev-cpp/glog
+	dev-cpp/gflags
+	>=dev-libs/boost-1.56.0
+	<dev-libs/boost-1.59.0
+	dev-libs/double-conversion
+	dev-libs/jemalloc
+	dev-libs/libevent
+	dev-libs/openssl:0
+	>=sys-devel/gcc-4.8.5:*
+	<sys-devel/gcc-5.1.0:*
```

Comment:
I suspect you're trying to enforce building with a specific gcc version. This won't work thanks to possible gcc-config switching, plus possible CC/CXX overrides. Furthermore, this kind of dependency will most likely install two gcc versions since you are allowing it to match any slot.

If you really need to restrict gcc version, you need to do it in `pkg_pretend`. However, before doing that please tell me what exactly you need the restriction for, so I can suggest the most correct solution.